core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -788,11 +779,15 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
     // FIXME_SYCL so far accessors used instead of these pointers
     // Functor's reduce memory, team scan memory, and team shared memory depend
     // upon team size.
-    const auto& space    = *m_policy.space().impl_internal_space_instance();
-    const sycl::queue& q = *space.m_queue;
-    m_scratch_ptr[0]     = nullptr;
-    m_scratch_ptr[1]     = sycl::malloc_device(
-        sizeof(char) * m_scratch_size[1] * m_league_size, q);
+    const auto& space = *m_policy.space().impl_internal_space_instance();
+    m_scratch_ptr[0]  = nullptr;
+    m_scratch_ptr[1]  = m_team_size <= 0
```

Comment:
I don't get the ternary operator here.  Did you mean to check if `m_league_size` is non negative?

Reply:
It doesn't matter here (but it did for the `Cuda` backend). Let me just remove it.

Reply:
272dd5597301e8568db473ddc8651b962d56c781